hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/LockManager.java

```diff
@@ -111,6 +111,7 @@ public void unlock() {
     if (writeConfig.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
       getLockProvider().unlock();
       metrics.updateLockHeldTimerMetrics();
+      close();
```

Comment:
Nice catch, can we write a UT to guard this logic?

Reply:
Is this because we reuse the client ? 

![image](https://user-images.githubusercontent.com/1179324/228163573-9fead5f9-56cf-45cd-90e9-b2f1714a9049.png)

Reply:
Of course, it depends on whether we want to keep the long live connection or short live connection, if we fix the code like this patch does, the connection becomes short lived.

A long lived connection is more risky for connection leaks, we need to care about the life cycle of the connection, we need to consider resuing the existing connection, which makes things more complicited and hard to keep correctness.

Reply:
@vinothchandar : yes you are right. for streaming ingestion, we keep re-using the same write client and hence.